### ISSUE TITLE
New issue_type: referential integrity between geography and document datasets

### ISSUE THREAD

**Issue by Ben-Hodgkiss** [digital-land/technical-documentation#86]
**Overview**

Data management need a new data quality assessment in the pipeline to detect whether the foreign key values in geography datasets match primary key values in document datasets, as is expected.

This should apply to:

- the foreign key `tree-preservation-order`  in `tree` and `tree-preservation-zone` geography datasets, should match to primary key `reference` in the `tree-preservation-order` document dataset (see [here for diagram](https://digital-land.github.io/specification/specification/tree-preservation-order/ "‌"))
- the foreign key `article-4-direction`  in the `article-4-direction-area` geography dataset,  should match to the primary key `reference`  in the `article-4-direction` document dataset (see [here for diagram](https://digital-land.github.io/specification/specification/article-4-direction/ "‌"))

‌

Optional (depending on tech approach below may make sense to split into a different issue type?)

- this could also include the `conservation-area` dataset, where the relationship is in reverse, and the primary key `reference` in the `conservation-area` geography dataset should match the foreign key `conservation-area` in the `conservation-area-document` document dataset.

**Tech Approach**
To be completed by dev.

**Acceptance Criteria/Tests**

- any foreign key value in a geography dataset which does not match to one of the primary keys in the document dataset should trigger a new issue record with issue_type = “no associated documents found for this area”
- this should only occur when a document dataset has been submitted by an LPA (e.g. if an LPA has only shared a `tree` dataset and not a `tree-preservation-order` dataset the issue check shouldn’t take place.
- the issue should be retrospectively applied to all pairs of geography and document datasets.
- the issue should be displayed when users check their data in the Publish/Check service. This may require a secondary ticket.

**Resourcing**
_Are there any tickets that need to be completed before this one can be? Are there any limitations as to who in the team can complete this ticket?_
